test/testextra/test_blake3.py

```diff
@@ -0,0 +1,172 @@
+      },
+      {
+        "input_len": 1023,
+        "hash": "10108970eeda3eb932baac1428c7a2163b0e924c9a9e25b35bba72b28f70bd11"
+      },
+      {
+        "input_len": 1024,
+        "hash": "42214739f095a406f3fc83deb889744ac00df831c10daa55189b5d121c855af7"
+      },
+      {
+        "input_len": 2048,
+        "hash": "e776b6028c7cd22a4d0ba182a8bf62205d2ef576467e838ed6f2529b85fba24a"
+      },
+      {
+        "input_len": 2049,
+        "hash": "5f4d72f40d7a5f82b15ca2b2e44b1de3c2ef86c426c95c1af0b6879522563030"
+      },
+      {
+        "input_len": 3072,
+        "hash": "b98cb0ff3623be03326b373de6b9095218513e64f1ee2edd2525c7ad1e5cffd2"
+      },
+      {
+        "input_len": 3073,
+        "hash": "7124b49501012f81cc7f11ca069ec9226cecb8a2c850cfe644e327d22d3e1cd3"
+      },
+      {
+        "input_len": 4096,
+        "hash": "015094013f57a5277b59d8475c0501042c0b642e531b0a1c8f58d2163229e969"
+      },
+      {
+        "input_len": 4097,
+        "hash": "9b4052b38f1c5fc8b1f9ff7ac7b27cd242487b3d890d15c96a1c25b8aa0fb995"
+      },
+      {
+        "input_len": 5120,
+        "hash": "9cadc15fed8b5d854562b26a9536d9707cadeda9b143978f319ab34230535833"
+      },
+      {
+        "input_len": 5121,
+        "hash": "628bd2cb2004694adaab7bbd778a25df25c47b9d4155a55f8fbd79f2fe154cff"
+      },
+      {
+        "input_len": 6144,
+        "hash": "3e2e5b74e048f3add6d21faab3f83aa44d3b2278afb83b80b3c35164ebeca205"
+      },
+      {
+        "input_len": 6145,
+        "hash": "f1323a8631446cc50536a9f705ee5cb619424d46887f3c376c695b70e0f0507f"
+      },
+      {
+        "input_len": 7168,
+        "hash": "61da957ec2499a95d6b8023e2b0e604ec7f6b50e80a9678b89d2628e99ada77a"
+      },
+      {
+        "input_len": 7169,
+        "hash": "a003fc7a51754a9b3c7fae0367ab3d782dccf28855a03d435f8cfe74605e7817"
+      },
+      {
+        "input_len": 8192,
+        "hash": "aae792484c8efe4f19e2ca7d371d8c467ffb10748d8a5a1ae579948f718a2a63"
+      },
+      {
+        "input_len": 8193,
+        "hash": "bab6c09cb8ce8cf459261398d2e7aef35700bf488116ceb94a36d0f5f1b7bc3b"
+      },
+      {
+        "input_len": 16384,
+        "hash": "f875d6646de28985646f34ee13be9a576fd515f76b5b0a26bb324735041ddde4"
+      },
+      {
+        "input_len": 31744,
+        "hash": "62b6960e1a44bcc1eb1a611a8d6235b6b4b78f32e7abc4fb4c6cdcce94895c47"
+      },
+      {
+        "input_len": 102400,
+        "hash": "bc3e3d41a1146b069abffad3c0d44860cf664390afce4d9661f7902e7943e085"
+      }
+    ]
+
+  def generate_input(self, length: int) -> bytes:
+    return bytes(i % 251 for i in range(length))
+
+  def test_official_vectors(self):
+    """Test against the official test vectors from: https://github.com/BLAKE3-team/BLAKE"""
+    for vector in self.vectors:
+      input_len = vector["input_len"]
+      expected = vector["hash"]
+      text = self.generate_input(input_len)
+      actual = blake3(text)
+      self.assertEqual(actual, expected)
+
+  def test_file_input(self):
```

Comment:
Huh, why are files involved?

Reply:
Gotcha, updated, removed a bunch of lines.